reset-pass/account-validator-tool/src/java/org/sakaiproject/accountvalidator/tool/controller/MainController.java

```diff
@@ -0,0 +1,1283 @@
+    private DeveloperHelperService developerHelperService;
+    
+    @Autowired
+    private AuthenticationManager authenticationManager;
+    
+    @Autowired
+    private HttpServletRequest httpServletRequest;
+    
+    @Autowired
+    private UsageSessionService usageSessionService;
+    
+    @Autowired
+    private EmailTemplateService emailTemplateService;
+    
+    @Autowired
+    protected SiteService siteService;
+    
+    @Autowired
+    protected AuthzGroupService authzGroupService;
+    
+    private static final String TEMPLATE_KEY_ACKNOWLEDGE_PASSWORD_RESET = "acknowledge.passwordReset";
+    private static final String MAX_PASSWORD_RESET_MINUTES = "accountValidator.maxPasswordResetMinutes";
+    private static final int MAX_PASSWORD_RESET_MINUTES_DEFAULT = 60;
+
+    private Locale userLocale;
+
+    ArrayList <String> errorMsgs = new ArrayList<String>();    
+
+    ValidationAccount va;
+    
+    private Locale localeResolver(HttpServletRequest request, HttpServletResponse response) {
+
+        String userId = sessionManager.getCurrentSessionUserId();
+        final Locale loc = StringUtils.isNotBlank(userId) ? preferencesService.getLocale(userId) : Locale.getDefault();
+        LocaleResolver localeResolver = RequestContextUtils.getLocaleResolver(request);
+        localeResolver.setLocale(request, response, loc);
+
+        return loc;
+    }
+    
+    // REQUEST ACCOUNT 
+        
+    @RequestMapping(value = {"/requestAccount"}, method = RequestMethod.GET)
+    public String reqAccIndex(Model model, HttpServletRequest request, HttpServletResponse response, 
+            @RequestParam(value="tokenId", required = false) String tokenId) {
+                   
+        userLocale = localeResolver(request, response);
+
+        model.addAttribute("uiService", serverConfigurationService.getString("ui.service", "Sakai"));
+        
+        // va = getValidationAccount(tokenId, model);
+        // getValidationAccount {
+        
+        if (tokenId == null || "".equals(tokenId)) {
+
+            log.debug("Valid token not found");
+            model.addAttribute("warnMsg", messageSource.getMessage("msg.noCode", null, userLocale));
+            return "error.html";
+        }
+              
+        va = validationLogic.getVaLidationAcountBytoken(tokenId);
+        
+        if (va == null) {
+
+            log.debug("Not a valid validation code");
+            model.addAttribute("warnMsg", messageSource.getMessage("msg.noSuchValidation", new String[]{tokenId}, userLocale));
+            return "error.html";
+        // } checks left:    
+        } else if (!va.getAccountStatus().equals(ValidationAccount.ACCOUNT_STATUS_REQUEST_ACCOUNT)) {
+            // this form is not appropriate, no such validation of the required account status
+            model.addAttribute("warnMsg", messageSource.getMessage("msg.noSuchValidation", new String[]{va.getValidationToken()}, userLocale));
+            return "error.html";
```

Comment:
Create a ERROR_TEMPLATE variable and use the variable for all the error.html templates.